File: Programming Language/C#/minimum-number-of-vertices-to-reach-all-nodes.cs
public class Solution {
    public IList<int> FindSmallestSetOfVertices(int n, IList<IList<int>> edges) {
        int [] inDegree = new int[n];
        for(int i = 0; i < n; i++){
            inDegree[i] = 0;
        }
        foreach(List<int> edge in edges){
            inDegree[edge[1]]++;
        }
        List<int> ansList = new List<int>();
        for(int i = 0; i < n; i++){
            if(inDegree[i] == 0){
                ansList.Add(i);
            }
        }
        return ansList;
    }
}

File: Programming Language/Python3/minimum-number-of-vertices-to-reach-all-nodes.py
class Solution:
    def findSmallestSetOfVertices(self, n: int, edges: List[List[int]]) -> List[int]:
        inDegree = [0]*n
        for lst in edges:
            inDegree[lst[1]] += 1
        lst = []
        for i in range(0, n):
            if inDegree[i] == 0:
                lst.append(i)
        return lst
